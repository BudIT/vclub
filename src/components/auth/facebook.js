import React, { Component, PropTypes } from 'react';

const FacebookLogin = () => {
  window.fbAsyncInit = () => {
    FB.init({
      appId: '157524328028332',
      cookie: true,
      xfbml: true,
      version: 'v2.8',
    });
  };



  function myFacebookLogin() {
    FB.getLoginStatus((response) => {
      if (response.status === 'connected') {
        console.log('Logged in.');
        FB.api('/me/feed', (response) => {
          console.log(JSON.stringify(response));
        });
      }
    });
  }

  return (
    <div id="fb-root">
      <button
        className="fb-login-button"
        data-max-rows="1"
        data-size="medium"
        data-show-faces="false"
        data-auto-logout-link="true"
        onClick={myFacebookLogin}
      />
    </div>
  );
};

(function (d, s, id) {
  const fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) return;
  const js = d.createElement(s); js.id = id;
  js.src = '//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.8&appId=157524328028332';
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));

export default FacebookLogin;

