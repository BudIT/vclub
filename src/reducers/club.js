import { combineReducers } from 'redux';
import { INITIALIZE } from 'vclub/constants/actionTypes';

import membersReducer from './club/members';
import dataReducer from './club/data';


const baseReducer = combineReducers({
  members: membersReducer,
  data: dataReducer,
});

export default function clubReducer(atom, action) {
  if (action.type === INITIALIZE) {
    const newAtom = action.payload;
    newAtom.data.authenticated = true;
    newAtom.data.authenticating = false;

    return newAtom;
  }

  return baseReducer(atom, action);
}
