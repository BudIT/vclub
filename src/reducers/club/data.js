import createReducer from 'vclub/redux/createReducer';

import { auth } from 'vclub/actions/core';


export default createReducer({
  [auth]: (atom) => ({
    ...atom,
    authenticating: true,
  }),
}, {
  authenticated: false,
  authenticating: false,
  currentRoom: 'chat',
});
