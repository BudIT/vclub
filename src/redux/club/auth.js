import actionCreator from 'borex-actions/actionCreator';
import commandCreator from 'borex-actions/commandCreator';
import withSideEffect from 'borex-actions/withSideEffect';
import createReducer from 'borex-reducers/createReducer';
import setIn from 'borex-reducers/setIn';


export const auth = actionCreator(
  withSideEffect((context, authData, remember = true) => {
    const { ioSocket, localStorage, Raven } = context;

    ioSocket.emit('auth', authData);

    Raven.setUserContext(authData);

    Raven.captureMessage('Authenticated', {
      level: 'info',
      logger: 'auth',
      extra: authData,
    });

    if (remember) {
      localStorage.setItem('storedAuth', JSON.stringify(authData));
    }
  }),
);

export const setRestoredData = actionCreator();

export const restoreAuth = commandCreator((context) => {
  const { dispatch, localStorage } = context;
  const authJSON = localStorage.getItem('storedAuth');

  if (authJSON) {
    dispatch(setRestoredData(JSON.parse(authJSON)));
  }
});

export const logOut = commandCreator(() => {
  document.location.reload();
});


export default createReducer(on => {
  on(auth, setIn('authenticating', () => true));
  on(setRestoredData, setIn('restored'));
});
