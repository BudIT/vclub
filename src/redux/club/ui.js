import initialState from 'vclub/redux/initialClubState';

export const TOGGLE_MEMBER_PANEL = 'club/ui/toggle-member-panel';
export const TURN_ON_MEMBER_PANEL = 'club/ui/turn-on-member-panel';

export function toggleMemberPanel() {
  return {
    type: TOGGLE_MEMBER_PANEL,
  };
}

export function turnOnMemberPanel() {
  return {
    type: TURN_ON_MEMBER_PANEL,
  };
}

export default function reducer(state, action) {
  switch (action.type) {
    case TOGGLE_MEMBER_PANEL:
      return {
        ...state,
        showMemberPanel: !state.showMemberPanel,
      };
    case TURN_ON_MEMBER_PANEL:
      return { 
        ...state,
        showMemberPanel: true,
      };
    default:
      return state || initialState.ui;
  }
}
