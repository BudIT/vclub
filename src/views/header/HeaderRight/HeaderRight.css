.ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: flex-end;
  background-color: #FFD00B;
  flex: 1;
}

.tab {
  position: relative;

  color: #33363A;
  text-decoration: none;
  text-align: center;
  font-size: 22px;
  font-family: 'Rubik', helvetica, sans-serif;

  padding: 14px 16px;
  transition: 0.2s;
  display: inline-block;

  background-color: inherit;
  cursor: pointer;

  border-style: none;
  outline: none;

  border-right: 2px solid transparent;
}

.tab:hover {
  background-color: #FFAA00;
}

.tabDisabled {
  color: rgba(51, 54, 58, 0.5);
  cursor: default;
}

.tabDisabled:hover {
  background-color: transparent;
}

.errorMark {
  position: absolute;
  top: 25px;
  left: 25px;
  color: white;
  background-color: red;
  border-radius: 50%;
  font-size: 16px;
  font-family: 'Rubik', helvetica, sans-serif;
  text-align: center;
  width: 22px;
  height: 22px;
  line-height: 20px;
}

.dropdownContent {
  display: none;
  position: absolute;
  background-color: #FFD00B;
  min-width: 200px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  right: 0px;
  z-index: 1;
}

.dropdown {}

.dropdownContent .tab:hover {
  background-color: #FFAA00;
  border-right: 2px solid #33363A;
}

.dropdown:hover .dropdownContent {
  display: flex;
  flex-direction: column;
}
