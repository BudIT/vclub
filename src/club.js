import React from 'react';
import io from 'socket.io-client';
import { render } from 'react-dom';
import { createStore, compose, applyMiddleware } from 'redux';

import { Provider } from 'react-redux';

import ClubLayout from 'vclub/views/clubLayout/ClubLayout';

import clientActionBroker from 'vclub/redux/middlewares/clientActionBroker';
import sideEffectProcessor from 'vclub/redux/middlewares/sideEffectProcessor';
import reducer from 'vclub/redux/clubReducer';
import initialState from 'vclub/redux/initialClubState';
import { restoreAuth } from 'vclub/redux/club/auth';


const ioSocket = io({ path: '/vclub-socket', forceNew: true });

const storeEnhancer = compose(
  applyMiddleware(
    clientActionBroker(ioSocket),
    sideEffectProcessor({ context: { ioSocket, localStorage } }),
  ),
  window.devToolsExtension ? window.devToolsExtension() : f => f
);

const store = createStore(reducer, initialState, storeEnhancer);

ioSocket.on('dispatch', action => store.dispatch(action));

store.dispatch(restoreAuth());

render((
  <Provider store={store}>
    <ClubLayout />
  </Provider>
), document.getElementById('AppRoot'));
