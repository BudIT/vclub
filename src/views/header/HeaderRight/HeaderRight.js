import React, { PropTypes } from 'react';

import compose from 'recompose/compose';
import withHandlers from 'recompose/withHandlers';
import setDisplayName from 'recompose/setDisplayName';

// actions
import { logOut } from 'vclub/redux/club/auth';
import { toggleMemberPanel } from 'vclub/redux/club/ui';

import SvgIcon from 'vclub/components/icons/SvgIcon';

import style from './HeaderRight.css';

import MediaButton from './MediaButton/MediaButton';
import ResultQuickVoteOf from './vote/ResultQuickVoteOf';

import usersIcon from './ic_group_black_24px.svg';


const enhance = compose(
  setDisplayName('HeaderRight'),
  withHandlers({
    onToggleMemberPanel: props => () => {
      props.dispatch(toggleMemberPanel());
    },
    onLogOut: props => () => {
      props.dispatch(logOut());
    },
  }),
);

export function HeaderRightComponent(props) {
  const {
    numberOfMembers,
    showMemberPanel,
    onLogOut,
    onToggleMemberPanel,
    user,
    dispatch,
  } = props;

  const displayVoteMenu = user.master;

  return (
    <ul className={style.ul}>
      <li>
        <MediaButton className={style.tab} />
      </li>
      {displayVoteMenu && (
        <li>
          <ResultQuickVoteOf
            className={style.tab}
            activeClassName={style.activeTab}
            dispatch={dispatch}
          />
        </li>
      )}
      <li>
        <button
          className={showMemberPanel ? style['tab--active'] : style.tab}
          onClick={onToggleMemberPanel}
        >
          <SvgIcon className={style.usersIcon} glyph={usersIcon} />
          {numberOfMembers}
        </button>
      </li>
      <li className={style.dropdown}>
        <button
          className={style.tab}
        >
          &#9776;
        </button>
        <div className={style.dropdownContent}>
          <button
            className={style.tab}
            onClick={onLogOut}
          >
            Выйти
          </button>
        </div>
      </li>
    </ul>
  );
}

HeaderRightComponent.propTypes = {
  numberOfMembers: PropTypes.number.isRequired,
  showMemberPanel: PropTypes.bool.isRequired,
  onLogOut: PropTypes.func.isRequired,
  onToggleMemberPanel: PropTypes.func.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    master: PropTypes.bool.isRequired,
  }).isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default enhance(HeaderRightComponent);
