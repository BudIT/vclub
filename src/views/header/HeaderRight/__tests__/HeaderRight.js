/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */
// jest.mock('react/lib/ReactDefaultInjection')

import React from 'react';
import { mount, shallow } from 'enzyme';
import { toggleMemberPanel } from 'vclub/redux/club/ui'
import { logOut } from 'vclub/redux/club/auth'
import HeaderRight from '../HeaderRight';

const numberOfMembers = 5;

test('HeaderRight calls passed function on click', () => {
  const dispatchSpy = jest.fn();
  const wrapper = mount(
    <HeaderRight
      numberOfMembers={numberOfMembers}
      dispatch={dispatchSpy}
    />
  );

  const buttonToggleMemberPanel = wrapper.find('button').first();
  buttonToggleMemberPanel.simulate('click');

  const buttonLogOut = wrapper.find('div.dropdownContent > button')
  buttonLogOut.simulate('click')

  expect(true).toBe(true)
  expect(dispatchSpy).toHaveBeenCalledTimes(2);
  // expect(dispatchSpy).toHaveBeenCalledWith(toggleMemberPanel(), logOut());
});
