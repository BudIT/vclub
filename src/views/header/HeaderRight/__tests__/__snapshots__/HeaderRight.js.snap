exports[`test <HeaderRight /> renders correctly 1`] = `
<ul
  className="ul">
  <li>
    <MediaButton
      className="tab" />
  </li>
  <li>
    <ResultQuickVoteOf
      activeClassName="activeTab"
      className="tab"
      dispatch={[Function]} />
  </li>
  <li>
    <button
      className="tab"
      onClick={[Function]}>
      <Icon
        className="usersIcon"
        glyph="test-file-stub" />
      5
    </button>
  </li>
  <li
    className="dropdown">
    <button
      className="tab">
      ☰
    </button>
    <div
      className="dropdownContent">
      <button
        className="tab"
        onClick={[Function]}>
        Выйти
      </button>
    </div>
  </li>
</ul>
`;
